### PR TITLE
[ONNX] Use deepcopy inputs for ONNX ort test cases

### PR DESCRIPTION
**Pull request by BowenBao**
Running models with inplace operators will change values of input tensors. 
Deepcopy input tensors each time to keep the original input tensors intact.

### CODE REVIEW

**Comment by BowenBao** [comment]
cc @lara-hdr @neginraoof @spandantiwari for review

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -1086,19 +1093,28 @@ def forward(self, x):
...
+        x = torch.randn(4, 2, 3, requires_grad=True)
+        self.run_test(InplaceAddModel(), x)
+
     def test_rsqrt(self):
         class RsqrtModel(torch.nn.Module):
             def forward(self, x):
                 return x.rsqrt()
 
-        x = torch.randn(4, 2, 3, requires_grad=True).to(dtype=torch.float64)
+        x = torch.randn(4, 2, 3, requires_grad=True, dtype=torch.float64)
```
Comment (lara-hdr) [inline test/onnx/test_pytorch_onnx_onnxruntime.py]: Were you getting a  "Only Tensors created explicitly by the user (graph leaves) support the deepcopy protocol at the moment" error before the change here?

If so, does it mean that we would only be able to call run_test with inputs on which we do not do any operation?

Comment (BowenBao) [reply test/onnx/test_pytorch_onnx_onnxruntime.py]: Yes, that was the exact error I saw.

Yes. If we really want to pass in inputs that have prior operations we could also possibly use clone() replacing deepcopy. it won't be as convenient to handle nested lists tensors. Is there an example use case that requires prior operations on inputs?

Comment (lara-hdr) [reply test/onnx/test_pytorch_onnx_onnxruntime.py]: no, I am ok with this, I was just curious.
I can't think of a case that would require prior operations. If we do get one, we could make this change (or maybe add a boolean for when to deepcopy), but for now, no need to make the code complex.

**Review by lara-hdr — APPROVED** [review]

**Pull request closed without merging**